docs/cookbook.md

```diff
@@ -0,0 +1,412 @@
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/datagrid/supplier.yml
+datagrid:
+    supplier:
+        #[...]
+        properties:
+            id: ~
+            edit_link:
+                type: url
+                route: pim_customentity_rest_get
+                params:
+                    - id
+                    - customEntityName
+        actions:
+            edit:
+                type:      navigate
+                label:     acme.supplier.this.edit
+                icon:      edit
+                link:      edit_link
+                rowAction: true
+```
+
+And add the following configuration to link the entity to its edition form :
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/custom_entities.yml
+custom_entities:
+    supplier:
+        options:
+            edit_form_extension: acme-supplier-edit-form
+```
+
+At this point, you should be able to dislay the edition form. But you should also see that all the fields are empty except the code. It’s because the entity has no normalizer, so we need to write one :
+
+```php
+<?php
+
+namespace Acme\Bundle\SupplierBundle\Normalizer;
+
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\scalar;
+use Acme\Bundle\SupplierBundle\Entity\Supplier;
+
+class SupplierNormalizer implements NormalizerInterface
+{
+    public function normalize($object, $format = null, array $context = array())
+    {
+        return [
+            'id' => $object->getId(),
+            'code' => $object->getCode(),
+            'name' => $object->getName(),
+        ];
+    }
+
+    public function supportsNormalization($data, $format = null)
+    {
+        return $data instanceof Supplier;
+    }
+
+}
+```
+
+The `normalize` method must return an array containing the reference data properties, with the property names as array keys and the property values as array values.
+
+Now declare the normalizer as a service :
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/services.yml
+services:
+    acme.normalizer.supplier:
+        class: Acme\Bundle\SupplierBundle\Normalizer\SupplierNormalizer
+        tags:
+            - {name: pim_serializer.normalizer, priority: 200}
+```
+
+The service must be tagged as “pim_serializer.normalizer” to be identified as a serializer. You now have to load the 
+services.yml file via the dependency injection :
+
+```php
+<?php
+
+namespace Acme\Bundle\SupplierBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\Extension;
+use Symfony\Component\DependencyInjection\Loader;
+use Symfony\Component\Config\FileLocator;
+
+class AcmeSupplierExtension extends Extension
+{
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.yml');
+    }
+}
+```
```

Comment:
From 295 to 313, I would remove this part because it is a basic thing in Symfony